v2/client.go

```diff
@@ -1553,7 +1610,7 @@ LOOP:
 				err = ErrPartitionPaused
 			case codes.ResourceExhausted:
 				// Indicates the end of a readonly partition has been reached.
-				err = ErrReadonlyPartition
+				err = ErrStreamExhausted
```

Comment:
This is an API change. Previously `ResourceExhausted` would only be returned if the end of a readonly partition had been reached; now it can happen when the stop position is reached. I added an extra error type for this, though that might not be necessary.

Reply:
I like the change as this error feels more appropriate, but it is a breaking API change so it will require a new major version.

Reply:
I guess one alternative would be to re-assign `ErrReadonlyPartition` to point to `ErrStreamExhausted`. It's hacky but would prevent existing logic from breaking?

Reply:
`ErrReadonlyPartition` is also returned when trying to publish to a read-only partition. So aliasing it to `ErrStreamExhausted` wouldn't really make sense in that case.

I'm happy to just return `ErrReadonlyPartition` for now, we can create an issue to include the new error type in the next major release.

Reply:
Yeah, I think that makes sense for now.